haystack/modeling/data_handler/dataloader.py

```diff
@@ -75,3 +86,15 @@ def __len__(self):
             return num_batches
         else:
             return super().__len__()
+
+    def _compute_max_number_of_labels(self, batch, tensor_names) -> int:
+        """
+        Compute the maximum number of labels in a batch.
+        Each example may have a different number of labels, depending on the number of answers.
+        """
+        max_num_labels = 0
+        for example in batch:
+            for name, tensor in zip(tensor_names, example):
+                if name == "labels" and tensor.ndim > 0:
+                    max_num_labels = max(max_num_labels, tensor.size(0))
+        return max_num_labels
```

Comment:
I think we can simplify this:
```suggestion
        try:
            labels_index = tensor_names.index("labels")
            max_num_labels = max([sample[labels_index].size(0) for sample in batch])
        except ValueError:
            max_num_labels = 0

        return max_num_labels
```

Reply:
I tried this solution but it seems to have different results.
To make it work, we would need to change it, as follows:
`max_num_labels=max(tensor.size(0) for example in batch for i,tensor in enumerate(example) if i==labels_index)`

But I think the first implementation was more readable and probably equally efficient.
For the moment, I don't change this part. WDYT?

Reply:
Okay, let's leave it like it its :)